Sources/Parsers/BBCodeParser.php

```diff
@@ -222,565 +226,95 @@ class BBCodeParser extends Parser
-		],
-		[
-			'tag' => 'u',
-			'before' => '<u>',
-			'after' => '</u>',
-		],
-		[
-			'tag' => 'url',
-			'type' => 'unparsed_content',
-			'content' => '<a href="$1" class="bbc_link" target="_blank" rel="noopener">$1</a>',
-			'validate' => __CLASS__ . '::urlValidate',
-		],
-		[
-			'tag' => 'url',
-			'type' => 'unparsed_equals',
-			'quoted' => 'optional',
-			'before' => '<a href="$1" class="bbc_link" target="_blank" rel="noopener">',
-			'after' => '</a>',
-			'validate' => __CLASS__ . '::urlValidate',
-			'disallow_children' => ['email', 'ftp', 'url', 'iurl'],
-			'disabled_after' => ' ($1)',
-		],
-		// Legacy (alias of [color=white])
-		[
-			'tag' => 'white',
-			'before' => '<span style="color: white;" class="bbc_color">',
-			'after' => '</span>',
-		],
-		[
-			'tag' => 'youtube',
-			'type' => 'unparsed_content',
-			'content' => '<div class="videocontainer"><div><iframe frameborder="0" src="https://www.youtube.com/embed/$1?origin={hosturl}&wmode=opaque" data-youtube-id="$1" allowfullscreen loading="lazy"></iframe></div></div>',
-			'disabled_content' => '<a href="https://www.youtube.com/watch?v=$1" target="_blank" rel="noopener">https://www.youtube.com/watch?v=$1</a>',
-			'block_level' => true,
-		],
+		['class' => 'SMF\BBCode\H1'],
+		['class' => 'SMF\BBCode\H2'],
+		['class' => 'SMF\BBCode\H3'],
+		['class' => 'SMF\BBCode\H4'],
+		['class' => 'SMF\BBCode\H5'],
+		['class' => 'SMF\BBCode\H6'],
+		['class' => 'SMF\BBCode\Html'],
+		['class' => 'SMF\BBCode\Hr'],
+		['class' => 'SMF\BBCode\I'],
+		['class' => 'SMF\BBCode\Img'],
+		['class' => 'SMF\BBCode\Iurl1'],
+		['class' => 'SMF\BBCode\Iurl2'],
+		['class' => 'SMF\BBCode\Justify'],
+		['class' => 'SMF\BBCode\Right'],
+		['class' => 'SMF\BBCode\Li'],
+		['class' => 'SMF\BBCode\List1'],
+		['class' => 'SMF\BBCode\List2'],
+		['class' => 'SMF\BBCode\List3'],
+		['class' => 'SMF\BBCode\Ltr'],
+		['class' => 'SMF\BBCode\Me'],
+		['class' => 'SMF\BBCode\Member'],
+		['class' => 'SMF\BBCode\Move'],
+		['class' => 'SMF\BBCode\NoBBC'],
+		['class' => 'SMF\BBCode\Nolink'],
+		['class' => 'SMF\BBCode\Php'],
+		['class' => 'SMF\BBCode\Pre'],
+		['class' => 'SMF\BBCode\Quote1'],
+		['class' => 'SMF\BBCode\Quote2'],
+		['class' => 'SMF\BBCode\Quote3'],
+		['class' => 'SMF\BBCode\Quote4'],
+		['class' => 'SMF\BBCode\Red'],
+		['class' => 'SMF\BBCode\Right'],
+		['class' => 'SMF\BBCode\Rtl'],
+		['class' => 'SMF\BBCode\S'],
+		['class' => 'SMF\BBCode\Shadow'],
+		['class' => 'SMF\BBCode\Size1'],
+		['class' => 'SMF\BBCode\Size2'],
+		['class' => 'SMF\BBCode\Sub'],
+		['class' => 'SMF\BBCode\Sup'],
+		['class' => 'SMF\BBCode\Table'],
+		['class' => 'SMF\BBCode\Td'],
+		['class' => 'SMF\BBCode\Time1'],
+		['class' => 'SMF\BBCode\Time2'],
+		['class' => 'SMF\BBCode\Tr'],
+		['class' => 'SMF\BBCode\Tt'],
+		['class' => 'SMF\BBCode\U'],
+		['class' => 'SMF\BBCode\Url1'],
+		['class' => 'SMF\BBCode\Url2'],
+		['class' => 'SMF\BBCode\White'],
+		['class' => 'SMF\BBCode\YouTube'],
```

Comment:
Shouldn't this be calls like:
```
['class' => SMF\BBCode\YouTube::class],
```

Strictly typed and allows for CI/CD pipelines to identify that it exists.